index.rst

```diff
@@ -55,78 +55,452 @@
+    - calls the :py:meth:`defineQuanta <SuperTask.defineQuanta>` method of each :py:class:`SuperTask` in the :py:class:`Pipeline` in sequence, accumulating a list of all quanta to be executed;
+    - constructs the *Quantum Graph* (see :ref:`preflight`), a bipartate directed acyclic graph with quantum vertices linked by the dataset vertices they produce and consume.
+
+#.  The Quantum Graph is passed to an ExecutionFramework, along with additional configuration for how the processing is to be performed (changes in this configuration must not change the outputs of the :py:class:`Pipeline` except to allow intermediate datasets to be elided).  The ExecutionFramework may be the same class as the PreFlightFramework (as in :py:class:`lsst.pipe.base.CmdLineTask`, which performs both roles), which makes this step a no-op.  It may also be a completely different class that may be run in an entirely different compute environment (via a serialized Quantum Graph).
+
+#.  The ExecutionFramework creates one or more output data repositories and records in them any repository-wide provenance (such as the :py:class:`Pipeline` configuration or software versions).
+
+#.  The ExecutionFramework walks the Quantum Graph according to the partial ordering it defines, and calls ``runQuantum`` on the appropriate concrete SuperTask for each quantum vertex.  Depending on the activator, the SuperTasks may be run directly in the same compute environment, or submitted to a workflow system for execution elsewhere (probably by translating the generic Quantum Graph to a format specific to a particular workflow system).  In some environments a temporary local data repository containing only the datasets consumed by a particular set of quanta may be created in scratch space to support execution in a context in which the original data repositories are not accessible, with output datasets similarly staged back to the true output data repositories.
+
+.. note::
+
+    The above procedure does not provide a mechanism for adding camera-specific overrides to the configuration.  I think this has to be part of the :py:class:`Pipeline` interface that's done in the first step, not something done later by PreFlightFrameworks.  That's especially true if we want to permit Pipelines that aggregate data from multiple cameras; in that case I think we'd need the Pipeline itself to hold the overrides for different cameras in addition to the defaults to avoid spurious provenance issues from having different configurations of the same Pipeline in a repo.  Given that different cameras might even change the SuperTasks we want in a Pipeline, we may need to make it possible to parameterize all of a Pipeline's definition on different Units of data (not just cameras, but filters).  I'm sure that's doable, but it is not currently supported by the :py:class:`Pipeline` API in this document.
```

Comment:
Yes, since writing this my thinking has evolved in the direction of just not supporting Pipelines with different per-camera configuration.  We do need to find some way to be able to run Pipelines that can aggregate data from multiple cameras, but I think we can demand that the multi-camera inputs be standardized by previous processing (which may involve running different Pipelines) before running a single multi-camera Pipeline with one configuration to combine them.